src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/RcTreeImporter.kt

```diff
@@ -0,0 +1,124 @@
+import org.wycliffeassociates.otter.jvm.persistence.database.AppDatabase
+import org.wycliffeassociates.otter.jvm.persistence.entities.ResourceLinkEntity
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.CollectionMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.ContentMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.LanguageMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.ResourceMetadataMapper
+import org.wycliffeassociates.resourcecontainer.ResourceContainer
+
+class RcTreeImporter(
+        private val database: AppDatabase,
+        private val collectionRepository: ICollectionRepository,
+        private val contentRepository: IContentRepository
+) : IRcTreeImporter {
+
+    override fun importResourceContainer(rc: ResourceContainer, rcTree: Tree, languageSlug: String): Completable {
+        return Completable.fromAction {
+            database.transaction { dsl ->
+                val language = LanguageMapper().mapFromEntity(database.getLanguageDao().fetchBySlug(languageSlug, dsl))
+                val helpRcTargetMetadata: ResourceMetadata? =
+                        when (rc.type()) {
+                            // TODO: Identifier shouldn't just be set to "ult"
+                            "help" -> ResourceMetadataMapper().mapFromEntity(
+                                    database.getResourceMetadataDao().fetchByIdentifier("ult", dsl),
+                                    language
+                            )
+                            else -> null
+                        }
+                val metadata = rc.manifest.dublinCore.mapToMetadata(rc.dir, language)
+                val dublinCoreFk = database.getResourceMetadataDao().insert(ResourceMetadataMapper().mapToEntity(metadata), dsl)
+                val ih = ImportHelper(dublinCoreFk, helpRcTargetMetadata, dsl)
+
+                ih.importCollection(null, rcTree)
+            }
+        }.subscribeOn(Schedulers.io())
+    }
+
+    inner class ImportHelper(val dublinCoreId: Int, val helpRcTargetMetadata: ResourceMetadata?, val dsl: DSLContext) {
+
+        fun importCollection(parentId: Int?, node: Tree) {
+            val collection = node.value
+            if (collection is Collection) {
+                val id = when (helpRcTargetMetadata) {
+                    null -> addCollection(collection, parentId)
+                    else -> findCollectionId(collection, helpRcTargetMetadata)
+                }
+                for (childNode in node.children) {
+                    importNode(id, childNode)
+                }
+            }
+        }
+
+        private fun findCollectionId(collection: Collection, helpRcTargetMetadata: ResourceMetadata): Int {
+            // TODO: What to do instead of blocking get?
+            return collectionRepository.getBySlugAndContainer(collection.slug, helpRcTargetMetadata)
+                    .blockingGet()
```

Comment:
I am fairly sure we don't want to just do a blockingGet() without catching an error (no such collection). I intend to change this